Core.DomainModel/ItSystemUsage/ItSystemUsage.cs

```diff
@@ -868,9 +850,52 @@ public void MarkAsDirty()
             LastChanged = DateTime.UtcNow;
         }
 
-        public bool CheckSystemValidity()
+        public ItSystemUsageValidationResult CheckSystemValidity()
+        {
+            var errors = new List<ItSystemUsageValidationError>();
+
+            var today = DateTime.UtcNow.Date;
+
+            var dateErrors = CheckDatesValidity(today).ToList();
+            var validAccordingToStatus = CheckLifeCycleValidity();
+            //If either one of the conditions pass the system is considered active
+            var isValid = dateErrors.Any() == false && validAccordingToStatus.IsNone;
+
+            errors.AddRange(dateErrors);
+            if (validAccordingToStatus.HasValue)
+            {
+                errors.Add(validAccordingToStatus.Value);
+            }
+
+            return new ItSystemUsageValidationResult(errors, isValid);
+        }
+
+        private IEnumerable<ItSystemUsageValidationError> CheckDatesValidity(DateTime todayReference)
+        {
+            if (Concluded == null && ExpirationDate == null)
+                yield break;
+
+            var today = todayReference.Date;
+            var startDate = (this.Concluded ?? today).Date;
+
+            //Valid yet?
+            if (today < startDate)
+            {
+                yield return ItSystemUsageValidationError.StartDateNotPassed;
+            }
+
+            //Expired?
+            if (!ExpirationDate.HasValue || today > ExpirationDate.Value.Date)
```

Comment:
your boolean logic is off here...
It says:
```
If expiration is not defined
   return ERRROR
if today > expiration date
  return ERROR
```
```
It should be
If expiration IS defined AND today > expiration date
   return ERRROR
```

Reply:
really you should just extract and enddate as you do with the start date and then check on that

`var endDate = (ExpirationDate ?? DateTime.Max).Date`

Will clean up your validation logic and make it easier to spot these kind of mistakes